### PR TITLE
File/datalake service doxygen comments and detail struct name refine.

### PR DESCRIPTION
**Pull request by katmsft**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp [hunk 1]
```diff
@@ -207,22 +207,19 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     constexpr static const char* HeaderRequestIsServerEncrypted = "x-ms-request-server-encrypted";
 
     // The value must be "filesystem" for all filesystem operations.
-    class FileSystemResourceType {
+    class FileSystemResource {
```
Comment (vhvb1989) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp:210]: Is this considered private/implementation as it is from the protocol layer?

We need changelog if not

**Review by vhvb1989 — COMMENTED** [review]
The name of the PR mention updating doxygen, but the changes are renaming types.

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp:210]: It is private

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_responses.hpp [hunk 2]
```diff
@@ -92,154 +200,459 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
...
       std::string Id;
+
+      /**
+       * The permissions of the ACL.
+       */
       std::string Permissions;
 
       /**
-       * @brief Creates an Acl based on acl input string.
+       * Creates an Acl based on acl input string.
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_responses.hpp:229]: Need @brief for functions

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp [hunk 3]
```diff
@@ -28,32 +28,103 @@
...
+       */
       std::string ContentType;
+
+      /**
+       * @brief The hash of the content.
+       */
       Storage::ContentHash ContentHash;
     };
 
     struct AclFailedEntry
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp:69]: Doc

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp [hunk 4]
```diff
@@ -28,32 +28,103 @@
...
     {
+      /**
+       * @brief The name of the failed entry.
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp:72]: No @brief for members

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp [hunk 5]
```diff
@@ -30,56 +30,90 @@
 namespace Azure { namespace Storage { namespace Files { namespace Shares {
 
   namespace Models {
+
+    /**
+     * @brief The common HTTP headers.
+     */
     struct FileHttpHeaders
     {
+
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp:39]: extra empty line

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp:39]: ditto for other structs

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**